internal/nearcache/repairing_task.go

```diff
@@ -0,0 +1,576 @@
+	// port of: com.hazelcast.internal.nearcache.impl.invalidation.RepairingTask#runAntiEntropy
+	rt.lg.Debug(func() string {
+		return "ReparingTask.runAntiEntropy"
+	})
+	// get a copy of the handlers, so we don't have to deal with sync.Map.
+	handlers := map[string]RepairingHandler{}
+	rt.handlers.Range(func(k, v interface{}) bool {
+		handlers[k.(string)] = v.(RepairingHandler)
+		return true
+	})
+	err := rt.invalidationMetaDataFetcher.fetchMetadata(ctx, handlers)
+	if err != nil {
+		return fmt.Errorf("ReparingTask: runAntiEntropy: %w", err)
+	}
+	return nil
+}
+
+// RepairingHandler is the port of: com.hazelcast.internal.nearcache.impl.invalidation.RepairingHandler
+type RepairingHandler struct {
+	name               string
+	metadataContainers []*MetaDataContainer
+	nc                 *NearCache
+	toNearCacheKey     func(key interface{}) (interface{}, error)
+	ss                 *serialization.Service
+	ps                 *cluster.PartitionService
+	lg                 ilogger.LogAdaptor
+}
+
+func NewRepairingHandler(name string, nc *NearCache, partitionCount int32, ss *serialization.Service, ps *cluster.PartitionService, lg ilogger.LogAdaptor) RepairingHandler {
+	mcs := make([]*MetaDataContainer, partitionCount)
+	for i := int32(0); i < partitionCount; i++ {
+		mcs[i] = NewMetaDataContainer()
+	}
+	sk := nc.Config().SerializeKeys
+	f := func(key interface{}) (interface{}, error) {
+		if sk {
+			return key, nil
+		}
+		return ss.ToObject(key.(serialization.Data))
+	}
+	return RepairingHandler{
+		name:               name,
+		nc:                 nc,
+		metadataContainers: mcs,
+		toNearCacheKey:     f,
+		ss:                 ss,
+		ps:                 ps,
+		lg:                 lg,
+	}
+}
+
+func (h RepairingHandler) Name() string {
+	return h.name
+}
+
+func (h RepairingHandler) GetMetaDataContainer(partition int32) *MetaDataContainer {
+	return h.metadataContainers[partition]
+}
+
+func (h *RepairingHandler) UpdateLastKnownStaleSequence(md *MetaDataContainer, partition int32) {
+	var lastReceived int64
+	var lastKnown int64
+	for {
+		lastReceived = md.Sequence()
+		lastKnown = md.StaleSequence()
+		if lastKnown >= lastReceived {
+			break
+		}
+		if md.CASStaleSequence(lastKnown, lastReceived) {
+			break
+		}
+	}
+	h.lg.Trace(func() string {
+		return fmt.Sprintf("stale sequences updated:[map=%s,partition=%d,lowerSequencesStaleThan=%d,lastReceivedSequence=%d]",
+			h.name, partition, md.StaleSequence(), md.Sequence())
+	})
+}
+
+// Handle handles a single invalidation.
+func (h *RepairingHandler) Handle(key serialization.Data, source, partition types.UUID, seq int64) error {
+	// port of: com.hazelcast.internal.nearcache.impl.invalidation.RepairingHandler#handle(com.hazelcast.internal.serialization.Data, java.util.UUID, java.util.UUID, long)
+	// Apply invalidation if it's not originated by local member/client.
+	// Since local Near Caches are invalidated immediately there is no need to invalidate them twice.
+	if source != partition {
```

Comment:
In java this if is comparing localUUid (client uuid) with source uuid

Reply:
Fixed at: https://github.com/hazelcast/hazelcast-go-client/pull/807/commits/22d0f4598f4b66079464377bdbe5e4aee0db7a09